pkg/loghttp/query.go

```diff
@@ -347,33 +348,63 @@ func ParseIndexStatsQuery(r *http.Request) (*RangeQuery, error) {
 }
 
 func ParseSeriesVolumeQuery(r *http.Request) (*RangeQuery, error) {
```

Comment:
I'm not sure we have to change anything here. Calling `ParseRangeQuery` with a custom limit should still be sufficient because we can handle forcing the step to 0 in the Request encoder/decoder.

Reply:
`ParseRangeQuery` does not work because of the step logic used in 2 places.

1. There are cases where this is called and we already have a 0 step, which causes an error to be thrown here:
```
	if result.Step <= 0 {
		return nil, errZeroOrNegativeStep
	}
```

2. There are other cases where this is called before step has been set. Step does not need to be present in the request as we hardcode it based on the route, however this also fails because of how `ParseRangeQuery` sets a default step:
```
func step(r *http.Request, start, end time.Time) (time.Duration, error) {
	value := r.Form.Get("step")
	if value == "" {
		return time.Duration(defaultQueryRangeStep(start, end)) * time.Second, nil
	}
	return parseSecondsOrDuration(value)
}
```